src/libexpr/search-path.hh

```diff
@@ -0,0 +1,101 @@
+#pragma once
+///@file
+
+#include <optional>
+
+#include "types.hh"
+#include "comparator.hh"
+
+namespace nix {
+
+/**
+ * A "search path" is a list of ways look for something, used with
+ * `builtins.findFile`.
+ */
+struct SearchPath
+{
+    /**
+     * A single element of a `SearchPath`. A key-value pair.
```

Comment:
I don't think we should call the prefix a key. A "key" should be sufficient to identify something, some sort of injectivity, which is not the case here.

Reply:
Find with me. Got a suggestion?

Reply:
I removed all key/value references.